src/main/java/gregtech/api/worldgen/populator/SurfaceRockPopulator.java

```diff
@@ -53,17 +50,13 @@ public void initializeForVein(OreDepositDefinition definition) {
     private static Set<Material> findUndergroundMaterials(Collection<IBlockState> generatedBlocks) {
         Set<Material> result = new HashSet<>();
         for (IBlockState blockState : generatedBlocks) {
-            Material resultMaterial;
-            if (blockState.getBlock() instanceof IFluidBlock || blockState.getBlock() instanceof BlockLiquid) {
-                Fluid fluid = FluidRegistry.lookupFluidForBlock(blockState.getBlock());
-                resultMaterial = fluid == null ? null : MetaFluids.getMaterialFromFluid(fluid);
-            } else {
+            Material resultMaterial = null;
+            if (!(blockState.getBlock() instanceof IFluidBlock) && !(blockState.getBlock() instanceof BlockLiquid)) {
```

Comment:
Why this logic change? You are now skipping fluid blocks entirely, which breaks the underground oil sprouts, or fluid springs with custom fluids. This one should be reverted, and still scan for fluid blocks.

You could short circuit the fluid registry lookup by checking if the fluid block is an instanceof GTFluidBlock and retrieving the material, if it exists, from that, but the check does still need to exist